src/player.cpp

```diff
@@ -7909,12 +7904,16 @@ void player::suffer()
 
         bool power_armored = is_wearing_power_armor(&has_helmet);
 
+        double rads;
         if ((power_armored && has_helmet) || worn_with_flag("RAD_PROOF")) {
-            radiation += 0; // Power armor protects completely from radiation
+            rads = 0; // Power armor protects completely from radiation
         } else if (power_armored || worn_with_flag("RAD_RESIST")) {
-            radiation += rng(0, localRadiation / 40) + rng(0, selfRadiation / 5);
+            rads = localRadiation / 100.0f + selfRadiation / 10.0f;
         } else {
-            radiation += rng(0, localRadiation / 16) + rng(0, selfRadiation);;
+            rads = localRadiation / 30.0f + selfRadiation / 3.0f;
+        }
+        if( rads > 0 && x_in_y( rads, 1 ) ) {
+            radiation++;
```

Comment:
Any particular reason we're capping radiation exposure at one rad per turn? This makes less radioactive areas more dangerous, but makes highly radioactive areas much safer.  You can literally never absorb more than one rad per turn, massively extending how long you can spend in highly radioactive areas.  A quick look through the code indicated that in certain areas you could absorb up to 5 rads a turn with no protection.